CMakeLists.txt

```diff
@@ -135,7 +135,7 @@ find_package(ZLIB REQUIRED)
 
 # try to find an installed boost or install locally with conan
 set(boost_VERSION "1.71")
-find_package(Boost ${boost_VERSION} QUIET)
+find_package(Boost ${boost_VERSION} EXACT QUIET)
```

Comment:
Oh yeah, this is messed up.. Basically I did a bit of OSRM work lately, which also switched to conan but uses boost v1.79.0. When I configured this fork, CMake picked up that version and after I did some cleanup in loki's `polygon_search` some tests failed and I couldn't see my mistake. Turns out boost::geometry must have a regression where the `within(line, poly)` also catches Linestrings which only intersect.. I'll comment in the test below and hope I can find the time to make sure it's really a regression and file a bug over at bg.

Reply:
@nilsnolde I also found this bug, and can be replicated here: https://gist.github.com/ismailsunni/f1b1640e1e93d5e52384c287778f70a0 

![image](https://user-images.githubusercontent.com/1421861/219006346-d0ef4603-3072-4828-ad28-55ad40c4edc2.png)

Line 3 should be within the polyon, but boost says no. Although boost says that the line is intersect. The result should be similar with line 1, 2, and 6.

Reply:
Ah awesome, thanks so much for putting together a test case @ismailsunni ! Look very legit to me when reading over (i.e. no floating point issues (you're using double), it's not trying to be spherical/geoidal etc). So, you also tried with boost v1.79 or similar?

Damn, I was sorta hoping it's us doing weird things with `within` of bg line instead of polygon/ring objects. But your test case seems to prove there's legitimate problems with boost's `within`. I didn't check your test thoroughly: are we sure it's the right winding? I keep forgetting what winding an external ring is supposed to be with bg, but I think counter-clockwise?! It might be worth it to throw a `bg::correct(poly)` in there just to make sure that's not the case, that'll correct the winding and closedness of rings.

If all that checks out, it'd be best to file a bug upstream and see what they say. It's only a minor problem for us really, we can enforce using a possibly bug-free boost 1.71 via conan and CMake. But it's prohibiting us from ever updating that version to past the version that introduced the potential bug.

Reply:
Also wondering if we shouldn't update this whole thing to use `geos` instead. We won't get rid of it for the time being, but @kevinkreiser recently did a major refactor of the whole admin stuff (and in fact he has an open PR to remove bg from that part entirely), and while using the C lib is somewhat ugly/hard, it might be time well spent.

I'm not saying we want to push that on you, I think for the sake of this PR it's fine for now to live with the possible bug in bg and require v1.71..

Reply:
long story short yes, we should drop boost for geos. boost is not ready for spherical coordinate systems at this time